lib/middleware/healthcheck/healthcheck.js

```diff
@@ -1,63 +1,35 @@
-require('@ministryofjustice/module-alias/register-module')(module)
-
-const request = require('request-promise-native')
-const { PORT, ROUTES } = require('~/fb-runner-node/constants/constants')
-
-let HEALTHCHECK_URL = ROUTES.healthcheck
-const TEST_URL = `http://localhost:${PORT}`
-let VALIDATE_RESPONSE_METHOD
-
-const init = (validateHealthcheck, options = {}) => {
-  if (options.url) {
-    HEALTHCHECK_URL = options.url
-  }
-  if (validateHealthcheck) {
-    VALIDATE_RESPONSE_METHOD = validateHealthcheck
-  }
-  return (req, res, next) => {
-    if (req.originalUrl === HEALTHCHECK_URL) {
-      if (options.fakeHealthcheck) {
-        res.json({
-          status: true,
-          content: {
-            statusCode: 200
-          }
-        })
-      } else {
-        healthcheck(req, res)
-      }
-    } else {
-      next()
-    }
-  }
-}
-
-const healthcheck = (req, res) => {
-  let status = true
-  let statusCode
-  request(TEST_URL)
-    .then(html => {
-      statusCode = 200
-      if (VALIDATE_RESPONSE_METHOD) {
-        status = VALIDATE_RESPONSE_METHOD(html)
-      }
-    })
-    .catch(err => {
-      status = false
-      statusCode = err.statusCode
-    })
-    .then(() => {
-      res.status(status ? 200 : 500)
-      res.json({
-        status,
-        content: {
-          statusCode
-        }
-      })
-    })
-}
-
-module.exports = {
-  init,
-  healthcheck
-}
+require('@ministryofjustice/module-alias/register-module')(module)
+
+const request = require('request-promise-native')
+const {
+  PORT
+} = require('~/fb-runner-node/constants/constants')
+
+const TEST_URL = `http://localhost:${PORT}`
+
+function init (validate = () => true) {

```

Comment:
I refactored this. The `healthcheck` function was being exported but wasn't being used elsewhere in the app -- we may as well keep the promise chain inside `init`

Also, it is an idiosyncrasy of AR's that he doesn't quite know how to write promises. In his version, one `then` step would save values to a variable scoped outside of the promise chain, while a second `then` would consume the variable. In this version, the value modified in one `then` is set to an object given as the argument to a second `then`. (The `catch` between them does the same.)